sql/core/src/main/scala/org/apache/spark/sql/execution/local/HashJoinNode.scala

```diff
@@ -73,10 +60,21 @@ case class HashJoinNode(
     }
   }
 
+  /** Sets the HashedRelation used by this node. */
+  protected def withHashedRelation(hashedRelation: HashedRelation): Unit = {
+    hashed = hashedRelation
+  }
+
+  /**
+   * For nodes that extends this, they can use doOpen to add operations needed in the open method.
```

Comment:
I think just the following is good:
`Custom open implementation to be overridden by subclasses.`